### DISCUSSION TITLE
How can I tell whether a driver supports "Execution Plan"

### DISCUSSION THREAD

**Discussion by hamscher** [dbeaver/dbeaver#19523] [Q&A]
I have the enterprise edition and a connection using odbc sql server 17 but sometimes it can produce a plan (for very simple queries) and other times says "not supported".  The page about 'query execution plan' says it depends on the driver.  How can i tell whether any given driver supports "Execution Plan" ?  Thanks!

**Comment by LonwoLonwo** [comment]
Hello @hamscher

Could you please provide the log files? You can find instructions here: https://dbeaver.com/docs/wiki/Log-files/. Thank you.

**Comment by hamscher** [comment]
here they are
￼

**Reply by LonwoLonwo** [reply]
Please provide it here, on this GitHub page. It doesn't work via e-mail.

**Comment by hamscher** [comment]
oh, sorry. here they are:
[hamscher-dbeaver-logs-2023-03-29.zip](https://github.com/dbeaver/dbeaver/files/11102464/hamscher-dbeaver-logs-2023-03-29.zip)

**Reply by LonwoLonwo** [reply]
Seems like a driver restriction.

Could you try to execute 

SET STATISTICS XML ON

Before your SELECT query, could you go ahead and execute your SELECT query? Does it return an extra tab with the plan info?

You need to do SET STATISTICS XML OFF after that. 

![2023-03-29 20_07_26-DBeaver Ultimate 23 0 1 - _master_ Script-101](https://user-images.githubusercontent.com/45152336/228615251-6660a93e-1897-42a1-879d-4d1a92846593.png)

**Reply by hamscher** [reply]
That helped a lot, because the error message now said "SHOWPLAN permission denied".  So I connected with elevated privileges and then everything works fine.   So it was just a misleading message, it was "not supported" for a non-admin user.  Thank you!!

**Reply by LonwoLonwo** [reply]
Thanks for the claryfication@hamscher
